src/gui/plugins/visualize_lidar/VisualizeLidar.cc

```diff
@@ -264,65 +261,31 @@ void VisualizeLidar::Update(const UpdateInfo &,
 
   std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
 
+  if (this->dataPtr->topicName.empty())
+    return;
+
   if (this->dataPtr->lidarEntityDirty)
   {
-    auto lidarURIVec = common::split(common::trimmed(
-                  this->dataPtr->lidarString), "::");
-    if (lidarURIVec.size() > 0)
+    std::string topic = this->dataPtr->topicName;
+    auto lidarEnt =
+        _ecm.EntityByComponents(components::SensorTopic(topic));
+    if (lidarEnt == kNullEntity)
     {
-      auto baseEntity = _ecm.EntityByComponents(
-          components::Name(lidarURIVec[0]));
-      if (!baseEntity)
-      {
-        gzerr << "Error entity " << lidarURIVec[0]
-            << " doesn't exist and cannot be used to set lidar visual pose"
-            << std::endl;
-        return;
-      }
-      else
-      {
-        auto parent = baseEntity;
-        bool success = false;
-        for (size_t i = 0u; i < lidarURIVec.size()-1; i++)
-        {
-          auto children = _ecm.EntitiesByComponents(
-                            components::ParentEntity(parent));
-          bool foundChild = false;
-          for (auto child : children)
-          {
-            std::string nextstring = lidarURIVec[i+1];
-            auto comp = _ecm.Component<components::Name>(child);
-            if (!comp)
-            {
-              continue;
-            }
-            std::string childname = std::string(
-                            comp->Data());
-            if (nextstring.compare(childname) == 0)
-            {
-              parent = child;
-              foundChild = true;
-              if (i+1 == lidarURIVec.size()-1)
-              {
-                success = true;
-              }
-              break;
-            }
-          }
-          if (!foundChild)
-          {
-            gzerr << "The entity could not be found."
-                  << "Error displaying lidar visual" <<std::endl;
-            return;
-          }
-        }
-        if (success)
-        {
-          this->dataPtr->lidarEntity = parent;
-          this->dataPtr->lidarEntityDirty = false;
-        }
-      }
+      if (topic[0] == '/')
+        topic = topic.substr(1);
+      lidarEnt =
+        _ecm.EntityByComponents(components::SensorTopic(topic));
+    }
+
+    if (lidarEnt == kNullEntity)
```

Comment:
Should we do something to avoid this message to show up over and over if we don't find the entity?

Reply:
Updated to print only once. I also added extra checks for null entity in case the lidar is deleted. [cf6ed2c](https://github.com/gazebosim/gz-sim/pull/2481/commits/cf6ed2cb55b521b2a73c714e065cff8635540d0b)